waku/v2/waku_relay/protocol.nim

```diff
@@ -120,6 +179,12 @@ method stop*(w: WakuRelay) {.async.} =
   debug "stop"
   await procCall GossipSub(w).stop()
 
+# rejects messages that are not WakuMessage
+proc validator(pubsubTopic: string, message: messages.Message): Future[ValidationResult] {.async.} =
+  let msg = WakuMessage.decode(message.data)
```

Comment:
**Optimization note:** 

You don't need to decode a message to know if the message is correct. Although typically Protocol buffer libraries perform both at the same time when decoding, validation and deserialization can be performed independently. For example, using nim-libp2p's protobuf library, you can check if the required fields are present without allocating memory.

Reply:
interesting, added a note here: dc70df97f859a4fa907db0be13bfd56d7a23dce8 for future optimizations.